docs/dependency_tracking.rst

```diff
@@ -0,0 +1,99 @@
+
+Dependency Tracking
+###################
+
+One of the biggest features of rapids-cmake is that is can track dependencies ( `find_package`, `cpm` )
+allowing for projects to easily generate `<project>-config.cmake` files with correct dependency requirements.
+In a normal CMake project public dependencies need to be recorded in two locations, the original ``CMakeLists.txt`` file and also the generated `<project>-config.cmake`. This dual source of truth increases
+developer burden, and adds a common source of error.
+
+``rapids-cmake`` is designed to remove this dual source of truth by expanding the concept of Modern CMake `Usage Requirements <https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-specification-and-usage-requirements>`_  to include external packages.
+This is done via the ``BUILD_EXPORT_SET`` ( maps to  `<BUILD_INTERFACE>` ) and ``INSTALL_EXPORT_SET`` ( maps to `<INSTALL_INTERFACE>` ) keywords on commands such as :cmake:command:`rapids_find_package` and :cmake:command:`rapids_cpm_find`.
+Lets go over an example on how these come together to make dependency tracking for projects super easy:
+
+.. code-block:: cmake
+
+  rapids_find_package(Threads REQUIRED
+    BUILD_EXPORT_SET example-targets
+    INSTALL_EXPORT_SET example-targets
+    )
+  rapids_cpm_find(nlohmann_json 3.9.1
+    BUILD_EXPORT_SET example-targets
+    )
+
+  add_library(example src.cpp)
+  target_link_libraries(example 
+    PUBLIC Threads::Threads
+      $<BUILD_INTERFACE:nlohmann_json::nlohmann_json>
+    )
+  install(TARGETS example
+    DESTINATION lib
+    EXPORT example-targets
+    )                 
+
+  set(doc_string [=[Provide targets for the example library.]=])
+  rapids_export(INSTALL example
+    EXPORT_SET example-targets
+    NAMESPACE example::
+    DOCUMENTATION doc_string
+    )
+  rapids_export(BUILD example
+    EXPORT_SET example-targets
+    NAMESPACE example::
+    DOCUMENTATION doc_string
+    )
+
+
+Tracking find_package
+*********************
+
+The :cmake:command:`rapids_find_package(<PackageName>)` combines the :cmake:command:`find_package` command and dependency tracking.
+In this example we are recording that the `Threads` package is required by both the export set `example-targets` for both the install and build configuration.
```

Comment:
```suggestion
This example records that the `Threads` package is required by both the export set `example-targets` for both the install and build configuration.
```
I also think maybe you need to either repeat the code here or add "via the `BUILD_EXPORT_SET` and `INSTALL_EXPORT_SET` parameters" or something to link the explanation above to the code...